src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,214 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { autobind } from '@shopify/javascript-utilities/decorators';
+import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
+import { findFirstFocusableNode } from '@shopify/javascript-utilities/focus';
+import { addEventListener, removeEventListener } from '@shopify/javascript-utilities/events';
+
+import DropdownItem from './DropdownItem';
+import { classNames } from '@shopify/react-utilities/styles';
+import { DROPDOWN } from '../ThemeIdentifiers';
+import * as baseTheme from './Dropdown.scss';
+import { Keys } from '../../types';
+import Button from '../Button';
+
+export type Direction = 'up' | 'down' | 'left' | 'right';
+
+export interface DropdownItemProps {
+  children?: React.ReactNode;
+  disabled?: boolean;
+  divider?: boolean;
+  header?: boolean;
+  onClick?(): void;
+}
+
+export interface Props {
+  disabled?: boolean;
+  children?: string;
+  direction?: Direction;
+  active: boolean;
+  activatorWrapper?: string;
+  DropdownItems: DropdownItemProps[];
+  onClose?(): void;
+  toggle?() : void;
+}
+
+export interface State {
+  selectedIndex: number
+}
+
+const getUniqueID = createUniqueIDFactory('Dropdown');
+
+export class Dropdown extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { 
+      selectedIndex: 0 
+    };
+  }
+
+  private activatorContainer: HTMLElement | null;
+
+  private id = getUniqueID();
+
```

Comment:
@shyamk9 : why are we keeping id & activatorContainer as global variable inside class, can we make it attach to 'this' and use it